### PR TITLE
Added LogFwk* logging levels to be used by the framework

### PR DESCRIPTION
**Pull request by Dr15Jones**
#### PR description:

- Added LogFwkInfo and LogFwkVerbatim with new level FWKINFO
- Refactored internals of MessageLogger to make it easier to maintain

This should make it easier to set a theshold such that INFO messages are ignored but the framework status messages are still kept.

#### PR validation:

Code compiles and framework unit tests pass.

### CODE REVIEW

**Comment by Dr15Jones** [comment]
NOTE: I decided to wait for another pull request to actually use the new threshold as many non-framework configurations are dependent upon how the default MessageLogger config is structured right now.

**Comment by makortel** [comment]
@cmsbuild, please test

**Comment by Dr15Jones** [comment]
please test

File: FWCore/MessageLogger/interface/MessageSender.h [hunk 1]
```diff
@@ -29,8 +18,10 @@ namespace edm {
 
   public:
     // ---  birth/death:
-    MessageSender() : errorobj_p() {}
+    MessageSender() = default;
     MessageSender(ELseverityLevel const& sev, std::string_view id, bool verbatim = false, bool suppressed = false);
+    MessageSender(MessageSender&&) = default;
+    MessageSender(MessageSender const&) = default;
```
Comment (makortel) [inline FWCore/MessageLogger/interface/MessageSender.h:24]: How about defaulting or deleting the copy and move assignment operators explicitly?

File: FWCore/MessageLogger/src/ELseverityLevel.cc [hunk 2]
```diff
@@ -150,6 +110,7 @@ namespace edm {
       ret[ELsev_zeroSeverity] = "--";
       ret[ELsev_success] = "Debug";  // 4/21/09 mf
```
Comment (makortel) [inline FWCore/MessageLogger/src/ELseverityLevel.cc:111]: Do you want to remove the comment? (I'm fine with either way)

File: FWCore/MessageLogger/src/ELseverityLevel.cc [hunk 3]
```diff
@@ -188,6 +150,7 @@ namespace edm {
       ret[ELsev_zeroSeverity] = "ELzeroSeverity   ";
       ret[ELsev_success] = "ELdebug          ";  // 4/21/09
```
Comment (makortel) [inline FWCore/MessageLogger/src/ELseverityLevel.cc:151]: Do you want to remove the comment? (I'm fine with either way)

**Comment by Dr15Jones** [comment]
NOTE: another small change is coming

File: FWCore/MessageLogger/interface/MessageSender.h (hunk 1 above)
Comment (Dr15Jones) [reply FWCore/MessageLogger/interface/MessageSender.h:24]: done

File: FWCore/MessageLogger/src/ELseverityLevel.cc (hunk 2 above)
Comment (Dr15Jones) [reply FWCore/MessageLogger/src/ELseverityLevel.cc]: done

File: FWCore/MessageLogger/src/ELseverityLevel.cc (hunk 3 above)
Comment (Dr15Jones) [reply FWCore/MessageLogger/src/ELseverityLevel.cc]: done

**Comment by Dr15Jones** [comment]
The original pull request tests have succeeded in compiling the code and are now running the unit tests. I propose letting that test at least get through the unit test phase before launching a new set of tests on the updated code.

**Comment by Dr15Jones** [comment]
Looking at the test log seems there were some failed tests
```
+ TEST_ERRORS='---> test unitTestsGroup_1 had ERRORS
---> test GeometryMTDGeometryBuilderTestDriver had ERRORS
---> test GeometryMTDNumberingBuilderTestDriver had ERRORS
---> test GeometryMTDCommonDataTestDriver had ERRORS'
```

**Comment by Dr15Jones** [comment]
please test

**Comment by santocch** [comment]
+1

**Comment by makortel** [comment]
+1

**Comment by Dr15Jones** [comment]
@cms-sw/upgrade-l2 it would be great if you could sign this

**Comment by kpedro88** [comment]
+upgrade

**Comment by qliphy** [comment]
+1

**Pull request merged**